lambda_functions/rds_init/lambda-handler.py

```diff
@@ -0,0 +1,51 @@
+import json
+import logging
+import os
+import sys
+
+import psycopg2
+from aws_lambda_powertools.utilities import parameters
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+# This lambda function is only meant to be run once during cdk initialization,
+# as post-db creation initialization script
+
+bde_host = os.environ["BDE_HOST"]
+bde_ro_user = os.environ["BDE_RO_USER"]
+# bde_ro_cred = parameters.get_secret(os.environ["BDE_RO_USER_CRED"])
+
+rds_fdw_host = os.environ["RDS_FDW_HOST"]
+rds_fdw_db = os.environ["RDS_FDW_DB"]
+
+rds_fdw_root_cred = parameters.get_secret(os.environ["RDS_FDW_ROOT"])
+
+rds_fdw_root_cred = json.loads(rds_fdw_root_cred)  # type: ignore
+rds_fdw_root_user = rds_fdw_root_cred["username"]  # type: ignore
+rds_fdw_root_pw = rds_fdw_root_cred["password"]  # type: ignore
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+try:
+    conn = psycopg2.connect(
```

Comment:
Is it possible to `with psycopg2.connect(…) as conn:` to avoid leaking it in case of another error somewhere?

Reply:
Have pulled this out into a function.

Reply:
That's not unconditionally closing the connection. I mean [this sort of calling](https://www.psycopg.org/docs/usage.html#with-statement), which ensures that the connection is closed whether there's an error or not.

Reply:
As per the [docs](https://www.psycopg.org/docs/usage.html#with-statement) 
>exiting the connection’s with block doesn’t close the connection, but only the transaction associated to it

I have added a `try`, `finally` clause to close the connection.